codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -158,17 +159,20 @@ public void incrementWindowSize(ChannelHandlerContext ctx, Http2Stream stream, i
             // Update the stream window and write any pending frames for the stream.
             FlowState state = state(stream);
             state.incrementStreamWindow(delta);
-            frameSent = false;
+            needFlush = false;
             state.writeBytes(state.writableWindow());
-            if (frameSent) {
+            if (needFlush) {
                 flush();
             }
         }
     }
 
     @Override
-    public ChannelFuture sendFlowControlledFrame(ChannelHandlerContext ctx, Http2Stream stream,
+    public ChannelFuture sendFlowControlledData(ChannelHandlerContext ctx, Http2Stream stream,
             ByteBuf data, int padding, boolean endStream, ChannelPromise promise) {
+        if (!sentFrame) {
```

Comment:
On the same theme as @nmittler's [comment](https://github.com/netty/netty/pull/3353/files#r23475320)...

Is this intended to be applied per stream?  I'm not sure it makes sense just to apply it 1 time globally to the connection.